packages/agreement-email-sender/src/services/agreementEmailSenderService.ts

```diff
@@ -1,229 +1,398 @@
+  }
+  return tenant;
+}
 
+function retrieveAgreementDescriptor(
+  eservice: EService,
+  agreement: Agreement
+): Descriptor {
   const descriptor = eservice.descriptors.find(
     (d) => d.id === agreement.descriptorId
   );
 
   if (!descriptor) {
     throw descriptorNotFound(agreement.eserviceId, agreement.descriptorId);
   }
+  return descriptor;
+}
+
+async function retrieveHTMLTemplate(
+  templateKind: AgreementEventMailTemplateType
+): Promise<string> {
+  const filename = fileURLToPath(import.meta.url);
+  const dirname = path.dirname(filename);
+  const templatePath = `/resources/templates/${templateKind}.html`;
+
+  try {
+    const htmlTemplateBuffer = await fs.readFile(
+      `${dirname}/..${templatePath}`
+    );
+    return htmlTemplateBuffer.toString();
+  } catch {
+    throw htmlTemplateNotFound(templatePath);
+  }
+}
+
+export function getFormattedAgreementStampDate(
+  agreement: Agreement,
+  stamp: keyof Agreement["stamps"]
+): string {
+  const stampDate = agreement.stamps[stamp]?.when;
+
+  if (stampDate === undefined) {
+    throw agreementStampDateNotFound(stamp, agreement.id);
+  }
+  return dateAtRomeZone(new Date(Number(stampDate)));
+}
+
+async function sendAgreementActivationEmail(
+  emailManager: EmailManager,
+  readModelService: ReadModelService,
+  agreementV2Msg: AgreementV2,
+  templateService: HtmlTemplateService,
+  templateKind: AgreementEventMailTemplateType,
+  logger: Logger,
+  sender: { label: string; mail: string },
+  consumerName: string,
+  producerName: string,
+  recepientsEmail: string[],
+  interopFeUrl?: string
+): Promise<void> {
+  const agreement = fromAgreementV2(agreementV2Msg);
+  const htmlTemplate = await retrieveHTMLTemplate(templateKind);
+
+  const activationDate = getFormattedAgreementStampDate(
+    agreement,
+    "activation"
+  );
 
+  const eservice = await retrieveAgreementEservice(agreement, readModelService);
+  const descriptor = retrieveAgreementDescriptor(eservice, agreement);
   const mail = {
     subject: `Richiesta di fruizione ${agreement.id} attiva`,
-    to: [producerEmail, consumerEmail],
+    to: recepientsEmail,
     body: templateService.compileHtml(htmlTemplate, {
       activationDate,
       agreementId: agreement.id,
       eserviceName: eservice.name,
       eserviceVersion: descriptor.version,
-      producerName: producer.name,
-      consumerName: consumer.name,
+      producerName,
+      consumerName,
+      interopFeUrl,
     }),
   };
 
-  logger.info(`Sending email for agreement ${agreement.id} activation`);
+  logger.info(
+    `Sending email for agreement ${agreement.id} activation (${emailManager.kind})`
+  );
   await emailManager.send(
     { name: sender.label, address: sender.mail },
     mail.to,
     mail.subject,
     mail.body
   );
-  logger.info(`Email sent for agreement ${agreement.id} activation`);
+  logger.info(
+    `Email sent for agreement ${agreement.id} activation (${emailManager.kind})`
+  );
```

Comment:
Note that here you don't have the try-catch as in the other two methods (rejection / submission)

Reply:
added in this commit 4214d9a37